### PR TITLE
Update custom theme docs with info for fonts

### PR DESCRIPTION
**Pull request by JamesAtIntegratnIO**
## Hey, I just made a Pull Request!

This PR updates documentation for the custom theme. It provides a couple examples for providing custom fonts.

First PR, be gentle 😉 

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/getting-started/app-custom-theme.md
Comment (awanlin) [inline docs/getting-started/app-custom-theme.md:232]:
```suggestion
To add custom fonts, you first need to declare the font style following the `@font-face` syntax from [Mui Typography](https://mui.com/material-ui/customization/typography/). After that you can then utilize the `styleOverrides` of `MuiCssBaseline` under components to set the default font. 
```

File: docs/getting-started/app-custom-theme.md
Comment (awanlin) [inline docs/getting-started/app-custom-theme.md:258]:
```suggestion
If you want to utilize different fonts, then you can set the top level `fontFamily` to what you want for your body, and then override `fontFamily` in `typography` to control fonts for various headings. 
```

**Review by awanlin — CHANGES_REQUESTED** [review]
Thanks for this @JamesAtIntegratnIO, just left some quick comments to help resolve the Vale errors. Will do a full review later when I have more time 👍

**Comment by JamesAtIntegratnIO** [comment]
@awanlin thanks for the quick review. Had to go bbq for the family. I'll get these updated tonight.

File: docs/getting-started/app-custom-theme.md
Comment (JamesAtIntegratnIO) [reply docs/getting-started/app-custom-theme.md]: oh. Just saw the commit suggestion button. Whoops. But thanks. All fixed.

Comment (awanlin) [reply docs/getting-started/app-custom-theme.md]: No worries 👍

File: docs/getting-started/app-custom-theme.md
Comment (awanlin) [inline docs/getting-started/app-custom-theme.md:258]: We probably want a space above this line

File: docs/getting-started/app-custom-theme.md
Comment (awanlin) [inline docs/getting-started/app-custom-theme.md:312]: Looks to be an extra space here

File: docs/getting-started/app-custom-theme.md
Comment (awanlin) [inline docs/getting-started/app-custom-theme.md:231]: We should add a space after the heading

**Review by awanlin — CHANGES_REQUESTED** [review]
This looks really good, many thanks again for this contribution @JamesAtIntegratnIO. Just mostly nits left around formatting

File: docs/getting-started/app-custom-theme.md
Comment (JamesAtIntegratnIO) [reply docs/getting-started/app-custom-theme.md:234]: Really, you could place them anywhere as long as you can reference them. That was just the convention I landed on. I was trying to figure out if there was a webpack requirement for the loaders to work, but couldn't find the reference.

Comment (JamesAtIntegratnIO) [reply docs/getting-started/app-custom-theme.md:234]: Do we have a code structure doc we should reference or also establish this as the convention?

Comment (JamesAtIntegratnIO) [reply docs/getting-started/app-custom-theme.md:234]: I see how they did it here. I'll follow suit.

https://backstage.io/docs/getting-started/app-custom-theme/#create-react-component

Comment (awanlin) [reply docs/getting-started/app-custom-theme.md:239]: Yes, that's true but for other's it may not be that obvious. I've often helped people on Discord who didn't realize they could just create a folder to put their theme into. We added a note to make it more clear. One of the things we always need to keep in mind is that many who use Backstage may not be experts in its tech stack. That was and is true for myself.

Comment (awanlin) [reply docs/getting-started/app-custom-theme.md:240]: > I see how they did it here. I'll follow suit.
> 
> https://backstage.io/docs/getting-started/app-custom-theme/#create-react-component

Perfect!

File: docs/getting-started/app-custom-theme.md
Comment (JamesAtIntegratnIO) [reply docs/getting-started/app-custom-theme.md:231]: space added

File: docs/getting-started/app-custom-theme.md
Comment (JamesAtIntegratnIO) [reply docs/getting-started/app-custom-theme.md]: added space

**Comment by JamesAtIntegratnIO** [comment]
> Thanks for the contribution! All commits need to be DCO signed before they are reviewed. Please refer to the the [DCO section in CONTRIBUTING.md](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin) or the [DCO](https://github.com/backstage/backstage/runs/23302966957) status for more info.

How do I resolve this one?

**Comment by JamesAtIntegratnIO** [comment]
> > Thanks for the contribution! All commits need to be DCO signed before they are reviewed. Please refer to the the [DCO section in CONTRIBUTING.md](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin) or the [DCO](https://github.com/backstage/backstage/runs/23302966957) status for more info.
> 
> How do I resolve this one?

I think I did it.

**Comment by awanlin** [comment]
You're going to want to run `yarn prettier:fix` in the folder with the file you updated, that should fix the CI failures

**Review by awanlin — APPROVED** [review]
Awesome contribution @JamesAtIntegratnIO! 🚀 

We'll need a review by the maintainers now.

**Review by vinzscam — APPROVED** [review]

**Comment by awanlin** [comment]
And voila your first contribution has been merged @JamesAtIntegratnIO! 🎉 

Note: the build is a bit slow so might take a while for this to show up here: https://backstage.io/docs/getting-started/app-custom-theme

**Pull request merged**
